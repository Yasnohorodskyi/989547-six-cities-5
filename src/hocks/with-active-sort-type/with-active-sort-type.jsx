import React from "react";

export const withActiveSortType = (Component) => {
  class WithActiveSortType extends React.PureComponent {
    constructor(props) {
      super(props);
      this.state = {
        isOpened: false
      };

      this.handleOpenerClick = this.handleOpenerClick.bind(this);
    }

    handleOpenerClick() {
      this.setState({
        isOpened: !this.state.isOpened
      });
    }

    render() {
      return <Component {...this.props} isOpened={this.state.isOpened} onOpenerClick={this.handleOpenerClick}/>;
    }
  }

  return WithActiveSortType;
};
