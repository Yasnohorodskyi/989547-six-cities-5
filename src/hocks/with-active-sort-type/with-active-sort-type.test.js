import React from "react";
import renderer from "react-test-renderer";
import propTypes from "prop-types";

import {withActiveSortType} from "./with-active-sort-type";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: propTypes.oneOfType([
    propTypes.arrayOf(propTypes.node),
    propTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withActiveSortType(MockComponent);

it(`with active sort type should render`, () => {
  const tree = renderer
  .create(
      <MockComponentWrapped>
        <React.Fragment />
      </MockComponentWrapped>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
