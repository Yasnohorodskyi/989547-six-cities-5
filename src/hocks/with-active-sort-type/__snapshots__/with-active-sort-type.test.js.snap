// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with active sort type should render 1`] = `<div />`;
